src/Player/States/Move.gd

```diff
@@ -33,7 +34,10 @@ func physics_process(delta: float) -> void:
 
 	# Rotation
 	if move_direction:
-		player.look_at(player.global_transform.origin + move_direction, Vector3.UP)
+		# To avoid dividing by 0, use 1 if rotation_interpolation equals 0
+		var interpolation_factor: float = 1 if rotation_interpolation == 0 else delta / rotation_interpolation
+		var target_direction: = player.transform.looking_at(player.global_transform.origin + move_direction, Vector3.UP)
+		player.transform = player.transform.interpolate_with(target_direction, interpolation_factor)
```

Comment:
The interpolation factor should take time into account. Also, to ensure that the rotation happens at a fixed speed, you need to calculate it this way:
```gdscript
var lerp_rate := 1 - pow(angular_interpolation_rate, delta)
player.transform = player.transform.interpolate_with(target_direction, lerp_rate)
```

For a detailed explanation: https://www.construct.net/en/blogs/ashleys-blog-2/using-lerp-delta-time-924

Reply:
The interpolation factor did factor in time but this is a better formula